### ISSUE TITLE
Why we get the square display for all sections? 

### ISSUE THREAD

**Issue by chongchenmath** [odlgroup/odl#961]
Example:

```python
import odl
import numpy as np

# 3D
discr = odl.uniform_discr([-300, -300, -100], [300, 300, 100], [300, 300, 100])

volume = odl.phantom.shepp_logan(discr, modified=True)

volume.show(indices=np.s_[volume.shape[0] // 2, :, :])
volume.show(indices=np.s_[:, volume.shape[1] // 2, :])
volume.show(indices=np.s_[:, :, volume.shape[-1] // 2])
```

**Comment by kohr-h** [comment]
I remember fighting with that one. At the time there was no ideal solution, maybe there is one now.

The ratio between the axes is controlled by the `aspect='auto'` option to the `imshow` command that the ODL `show` method [uses](https://github.com/odlgroup/odl/blob/d5819838593bc2d74af90424c49c69dc3ba90ad2/odl/util/graphics.py#L239).

 For available options for `aspect` see [here](http://matplotlib.org/api/_as_gen/matplotlib.axes.Axes.set_aspect.html#matplotlib.axes.Axes.set_aspect). You can test passing `aspect='equal'` or `aspect=factor`, where `factor` is the factor by which the y axis will be stretched (`factor > 1`) or contracted (`factor < 1`).
If that doesn't work you can do
```python
import matplotlib.pyplot as plt
fig = volume.show(...)  # insert parameters
ax = plt.gca()
ax.set_aspect(...)
```

Just for the record, I think it's hard to find a setting that fits all purposes. In particular if you plot things like sinograms  where the axes can live on vastly different scales.

**Comment by chongchenmath** [comment]
Thanks a lot. It works. But the colorbar is not proportional to the length of the box. Do you @kohr-h know how to adjust it? 
 
Example:

```
import odl
import numpy as np

# 3D
discr = odl.uniform_discr([-300, -300, -100], [300, 300, 100], [300, 300, 100])

volume = odl.phantom.shepp_logan(discr, modified=True)

volume.show(indices=np.s_[volume.shape[0] // 2, :, :], aspect='equal')
volume.show(indices=np.s_[:, volume.shape[1] // 2, :], aspect='equal')
volume.show(indices=np.s_[:, :, volume.shape[-1] // 2], aspect='equal')
```

**Comment by adler-j** [comment]
In general, as in issue #905 (more reading there) the idea with show is to quickly provide "something useful" rather than be a standard plotting rutine for all use-cases.

If you want to use more advanced plotting, we usually recommend going full matplotlib.

**Comment by kohr-h** [comment]
> Do you @kohr-h know how to adjust it?

No, but here's how you get at it:
```python
fig = volume.show(...)
ax = fig.gca()
cbar = ax.images[0].colorbar
```
Then google for "matplotlib scale colorbar" or something.

**Comment by MatthiasJE** [comment]
I had some trouble with beautiful colorbars as well. Here is my solution for this (after googling ""matplotlib scale colorbar" or something." :) )

I have not found a case where this gives me something ugly. It always looks quite nice.
```
from matplotlib import pyplot as plt
from mpl_toolkits.axes_grid1 import make_axes_locatable

ax = fig.add_subplot('111')
sbplt = ax.imshow(x)
divider = make_axes_locatable(ax)
# create an axes on the right side of ax. The width of cax will be 5%
# of ax and the padding between cax and ax will be fixed at 0.05 inch.
cax = divider.append_axes('right', size='5%', pad=0.1)
plt.colorbar(sbplt, cax=cax)
```

**Comment by kohr-h** [comment]
I'll close this since the question has been answered.
